src/pcapi/core/offers/models.py

```diff
@@ -326,20 +327,27 @@ def offerType(self) -> Optional[dict]:
         for possible_type in all_types:
             if str(possible_type) == self.type:
                 return possible_type.as_dict()
-        # FIXME (dbaty, 2020-12-03): shouldn't we raise an error such as
-        #     raise ValueError(f"Unexpected offer type '{self.type}'")
-        # instead of returning None?
-        return None
+
+        logger.error("Offer with id %s has unexpected type '%s'", self.id, self.type)
+        raise ValueError(f"Unexpected offer type '{self.type}'")
```

Comment:
Est-ce qu'il y a un intérêt à logger **et** lever une erreur ? Il est très probable que la ValueError ne soit pas récupérée, et donc on va se retrouver avec 2 entrées dans le Sentry (une pour le `logger.error` et une pour la `ValueError`). Proposition :



```suggestion
        raise ValueError(f"Unexpected offer type '{self.type} for offer {offer.id}'")
```